[FTR] Refactor test/common/services/* -> packages/kbn-ftr-common-functional-[ui-]services/*

wayneseymour opened the pull request:
## Summary

Moving common services to respective new homes.

Resolves: https://github.com/elastic/kibana/issues/188541

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour committed:
[FTR] Refactor test/common/services

Resolves: https://github.com/elastic/kibana/issues/188541

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour committed:
satisfy "node scripts/build_api_docs --plugin @kbn/ftr-common-functional-services --stats comments --stats exports"
...specifially the "--stats exports" bit

wayneseymour committed:
fixup threat intel svcs

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour commented:
/ci

wayneseymour committed:
drop the junit:merge to see if this lessens the total ci time, such that the cypress tests finish

wayneseymour commented:
/ci

wayneseymour committed:
fixup yet another pointer for common svcs

wayneseymour committed:
try to drop cypress tests for now

wayneseymour commented:
/ci

wayneseymour committed:
put this back

wayneseymour commented:
/ci

wayneseymour committed:
drop these, just like I dropped defend_workflows script

wayneseymour commented:
/ci

wayneseymour committed:
Turn defend_workflows.sh and defend_workflows.sh back on in ci

wayneseymour commented:
/ci

dmlemeshko commented on packages/kbn-ftr-common-functional-services/services/index_patterns.ts:
```diff
@@ -6,10 +6,14 @@
+// The fallowing two constant values live in
+// '@kbn/data-views-plugin/server/constants' and
+// '@kbn/core-http-common' respectively.
+const INITIAL_REST_VERSION = '2023-10-31' as const;
+const ELASTIC_HTTP_VERSION_HEADER = 'elastic-api-version' as const;
```
_question_: Any reasons not to import it from original packages?

wayneseymour replied:
I think I left that as a placeholder but failed to add a "TODO" comment. 
Changed it back and running the type checker locally now, after I ran ` node scripts/lint_ts_projects --fix`

wayneseymour commented:
@MadameSheema I've tried locally and on ci and the these two scripts always timeout, usually with zero errors.
`.buildkite/scripts/steps/functional/defend_workflows.sh`
&&
`.buildkite/scripts/steps/functional/threat_intelligence.sh`

Is there anything I can add or change to fix this?

wayneseymour committed:
Fix up from Dima's comment;

Type checker passed locally with:
"info [tsc] exited with 0 after 893.5 seconds"

wayneseymour committed:
attempt to extend the timeout's of the two scripts that are being cancelled

wayneseymour commented:
/ci

wayneseymour committed:
put these back as it made no difference

wayneseymour commented:
/ci

wayneseymour committed:
[pairing with tom] drop svcs for now

wayneseymour commented:
/ci

wayneseymour committed:
[pairing with tom] add test/common/services back, but using the new 'homes' of the services

wayneseymour commented:
/ci

wayneseymour committed:
[pairing with tom] repoint to *

wayneseymour commented:
/ci

wayneseymour commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into ftr/refactor-test-common-services

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

davismcphee reviewed (approved):
Code-only review. Data Discovery typings file change LGTM 👍

opauloh reviewed (approved):
Cloud Security Posture changes LGTM

PhilippeOberti reviewed (approved):
LGTM for the Threat Hunting Investigations team

jennypavlova reviewed (approved):
Obs Ux Infra Services changes LGTM (code review only)

TinaHeiligers reviewed (approved):
LGTM

dmlemeshko commented on packages/kbn-ftr-common-functional-ui-services/services/security/system_indices_user.ts:
_question_: shouldn't we put `security` service in `kbn-ftr-common-functional-services` ?
I don't think it has any dependency on browser / Kibana UI and is used in both api_integration and functional test.

wayneseymour replied:
👀

dmlemeshko replied:
nvm, `TestUser` services is relying on Browser service.

wayneseymour replied:
You're fast dude!  I was just copying the link to that! lol

dmlemeshko reviewed (approved):
Code LGTM. Great progress!

jbudz commented:
This was reverted in https://github.com/elastic/kibana/pull/191765.

See https://buildkite.com/elastic/kibana-pull-request/builds/230868#01919ed7-15d5-425c-9b8e-146ed5fe9daf for the build failure.
